src/PushNotifications.Delivery.Buffered/InMemoryBufferedDelivery.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using PushNotifications.Contracts;
+using PushNotifications.Contracts.PushNotifications.Delivery;
+
+namespace PushNotifications.Delivery.Buffered
+{
+    public class InMemoryBufferedDelivery<T> : IDisposable, IPushNotificationDelivery where T : IPushNotificationBufferedDelivery
+    {
+        readonly T delivery;
```

Comment:
why not IPushNotificationBufferedDelivery?

Reply:
Interface is changed to IPushNotificationDeliveryCapableOfSendingMoreThenOneNotificationAtOnce so it is not confusing